terraform/eval_context_builtin.go

```diff
@@ -297,7 +297,11 @@ func (ctx *BuiltinEvalContext) EvaluationScope(self addrs.Referenceable, keyData
 		InstanceKeyData: keyData,
 		Operation:       ctx.Evaluator.Operation,
 	}
-	return ctx.Evaluator.Scope(data, self)
+	scope := ctx.Evaluator.Scope(data, self)
+	if mc := ctx.Evaluator.Config.DescendentForInstance(ctx.PathValue); mc != nil {
```

Comment:
Could there be a comment added as to what is happening here? This seems like a change to where experiments must be declared? (can be declared by a parent)? If this is accurate, the docs need to be updated. If I'm not being accurate here, a comment would help clarify this :D

Reply:
`DescendentForInstance` selects the specific module that matches `ctx.PathValue`, so it's checking against the module containing the expression we're currently evaluating, to match with how the experiment checking in the `configs` package works.

With that said, I'll still add a comment saying the same thing. :grinning: